### PR TITLE
First version to support Agent profiles

### PR DESCRIPTION
**Pull request by davidor**
### What does this PR do?

First version to support Agent profiles.

Only the basic scenarios are supported for now. I've left many TODO comments in the code with all the things that should be addressed.

Current limitations (most of them should be mentioned in the TODO comments):
- Profiles with more than one rule might not work properly (we probably need to change the approach used to generate the "default" profile".
- Conflicts between profiles are not handled.
- We need to watch nodes in the controller (and only generate events when there are changes in the labels).
- Only DaemonSets are supported for now. We should do the same for EDS.
- We need integration tests.
- We should add something in the profile status field.
- I don't think `lt` and `gt` operators work well because they're not exact opposites.
- Only resource overrides are supported for now. It's what we have in the CRD, but we should add features at some point.

### CODE REVIEW

**Review by fanny-jiang — COMMENTED** [review]
LGTM. I'm still testing manually

File: pkg/agentprofile/agent_profile.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+// DaemonSetName returns the name that the DaemonSet should have according to
+// the name of the profile associated with it.
+func DaemonSetName(profileName string) string {
+	if profileName == defaultProfileName {
+		return "" // Return empty so it does not override the default DaemonSet name
+	}
+
+	return daemonSetNamePrefix + profileName
```
[the remark this answers is not in the document]
Comment (davidor) [reply pkg/agentprofile/agent_profile.go:59]: Right. Given that it's a small change we can do it now to avoid problems in the future.

**Review by fanny-jiang — APPROVED** [review]
LGTM

File: pkg/agentprofile/agent_profile.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidor) [reply pkg/agentprofile/agent_profile.go]: I've added a new commit with this.
I also added a TODO in the code with something related with this that I think will help us simplify a bit the code.

**Review by fanny-jiang — APPROVED** [review]

**Pull request merged**